### ISSUE TITLE
possible erroneous logging of werror

### ISSUE THREAD

**Issue by apache-hb** [mesonbuild/meson#8620]
**Describe the bug**
when building a simple project with `default_options : [ 'werror=true' ]` meson is logging 
```
Option werror is: true [default: true]
```
I believe this is a regression as versions prior to `0.57` didnt emit this and it seems redundant to log that a default is being used

**To Reproduce**
```
project('bug', 'c', default_options : [ 'werror=true' ])
```

**Expected behavior**
```
Option werror is: true [default: true]
```
is omitted from stdout

**system parameters**
meson version `0.57.1`

**Comment by tristan957** [comment]
cc @eli-schwartz I think this can be closed. I have never noticed this.

**Comment by eli-schwartz** [comment]
This was introduced in 9620755ff7db9c2614ea1929fcc95bec22734cd0 and deleted in 5e55a0bb2b95546da41fd27dbccc8c056815f391. The new approach is... considerably... more accurate.

Fixed in #9203
